### PR TITLE
Semantic conventions for browser events

### PR DESCRIPTION
**Pull request by scheler**
Introduces semantic conventions for events emitted by browser instrumentations. This document serves more for the purpose as a schema for the events than the semantic conventions as supported in Otel today.

Related PRs https://github.com/open-telemetry/opentelemetry-specification/pull/2926

### CODE REVIEW

**Comment by joaopgrassi** [comment]
Hey @scheler the semantic conventions moved to a new repo, https://github.com/open-telemetry/semantic-conventions so you will need to submit your PR there. 

Also, the conventions need to be introduced in a `.yaml` file, so the table and etc is auto-generated and populated in the markdown file.

File: specification/logs/events/semantic_conventions/browser.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+**[1]:**  Alias for [`http.url`](../../../trace/semantic_conventions/http.md)
+**[2]:** The URL fragment may be included for virtual pages
+
+`type` MUST be one of the following:
+
+| Value  | Description |
+|---|---|
+| `0` | physical_page - Initial page load within the browser and will generally also precede a PageNavigationTiming event.|
+| `1` | virtual_page - This is for Single Page Applications (SPA) where the framework provides the ability to perform client side only page "navigation", the exact definition of what a virtual page change is determined by the SPA and the framework it is using.|
+
```
Comment (Abinet18) [inline specification/logs/events/semantic_conventions/browser.md:58]: Should we have an attribute to show the change in state that happened on soft navigation (virtual pages) like pushState and replaceState. Name can be changeState

File: specification/logs/events/semantic_conventions/browser.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+| domContentLoadedEventEnd | long | || Recommended |
+| domComplete | long | || Recommended |
+| loadEventStart | long | || Recommended |
+| loadEventEnd | long | || Recommended |
+| firstPaint | long | || Recommended |
+| firstContentfulPaint | long | || Recommended |
+
+### ResourceTiming
+
+**Event name**:`resource_timing`
```
Comment (jsuereth) [inline specification/logs/events/semantic_conventions/browser.md:139]: Shouldn't this include an attribute denoting the resource that was fetched?

File: specification/logs/events/semantic_conventions/browser.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+```
+
+</details>
+
+### PageNavigationTiming
+
+**Event name**:`page_navigation_timing`
+
+This event describes the timing metrics of a page navigation as provided by
+`PerformanceNavigationTiming` Performance API.
```
Comment (jsuereth) [inline specification/logs/events/semantic_conventions/browser.md:121]: Does this include the page that is loading somewhere?

File: specification/logs/events/semantic_conventions/browser.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+|connectStart | long | || Recommended |
+|secureConnectionStart | long | || Recommended |
+|connectEnd | long | || Recommended |
+|requestStart | long | || Recommended |
+|responseStart | long | || Recommended |
+|responseEnd | long | || Recommended |
+
+### Exception
+
+**Event name**:`exception`
```
Comment (jsuereth) [inline specification/logs/events/semantic_conventions/browser.md:159]: Why is this redefined?  Shouldn't this just leverage existing exception specification, e.g. https://github.com/open-telemetry/semantic-conventions/blob/main/specification/logs/semantic_conventions/exceptions.md

File: specification/logs/events/semantic_conventions/browser.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+`user_action_type` MUST be one of the following:
+
+| Value  | Description |
+|---|---|
+| `click` | click |
+
+### WebVital
+
+**Event name**:`web_vital`
```
Comment (jsuereth) [inline specification/logs/events/semantic_conventions/browser.md:190]: Can you outline in more detail what this is?

File: specification/logs/events/semantic_conventions/browser.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+| `exception.line_number` | int | Line number where the error occurred |  | Recommended |
+| `exception.column_number` | int | Column number where the error occurred |  | Recommended |
+| `exception.message` | string | The exception message. | `Division by zero`; `Can't convert 'int' object to str implicitly` | Recommended |
+| `exception.stacktrace` | string | A stacktrace as a string in the natural representation for the language runtime. The representation is to be determined and documented by each language SIG. | `Exception in thread "main" java.lang.RuntimeException: Test exception\n at com.example.GenerateTrace.methodB(GenerateTrace.java:13)\n at com.example.GenerateTrace.methodA(GenerateTrace.java:9)\n at com.example.GenerateTrace.main(GenerateTrace.java:5)` | Recommended |
+| `exception.type` | string | The type of the exception (its fully-qualified class name, if applicable). The dynamic type of the exception should be preferred over the static type in languages that support it. | `java.net.ConnectException`; `OSError` | Recommended |
+
+### UserAction
+
+**Event name**:`user_action`
```
Comment (jsuereth) [inline specification/logs/events/semantic_conventions/browser.md:172]: Can you outline in more detail what user actions should be captured?

File: specification/logs/events/semantic_conventions/browser.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
+# Semantic conventions for browser events
```
Comment (jsuereth) [inline specification/logs/events/semantic_conventions/browser.md:1]: This should make use of semantic conventions YAML definitions for attributes, where possible.

File: specification/logs/events/semantic_conventions/browser.md (hunk 1 above)
Comment (scheler) [reply specification/logs/events/semantic_conventions/browser.md:58]: Added.

File: specification/logs/events/semantic_conventions/browser.md (hunk 2 above)
Comment (scheler) [reply specification/logs/events/semantic_conventions/browser.md:140]: Added `name` property that denotes the URL of the resource fetched. This is as per https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/name

File: specification/logs/events/semantic_conventions/browser.md (hunk 3 above)
Comment (scheler) [reply specification/logs/events/semantic_conventions/browser.md:121]: Added `name` property that denotes the URL of the page loaded. This is as per https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/name

File: specification/logs/events/semantic_conventions/browser.md (hunk 5 above)
Comment (scheler) [reply specification/logs/events/semantic_conventions/browser.md:194]: Added a short description for this event type.

File: specification/logs/events/semantic_conventions/browser.md (hunk 6 above)
Comment (scheler) [reply specification/logs/events/semantic_conventions/browser.md:174]: Added a few user actions captured using this Event.

**Comment by jsuereth** [comment]
Can you please reopen this against the current main branch, including using `yaml` files to define schema?

**Pull request closed without merging**